src/form-builder.tsx

```diff
@@ -23,50 +18,36 @@ export const FormBuilder = () => {
     // EFFECTS
     // =========================================================================
     useEffect(() => {
-        const handleResize = () => {
-            setIsLargeScreen(window.innerWidth >= 1200);
-        };
-
-        window.addEventListener("resize", handleResize);
-        return () => {
-            window.removeEventListener("resize", handleResize);
-        };
+        if (window) {
+            const handleResize = () => {
+                setIsLargeScreen(window.innerWidth >= 1200);
+            };
+            window.addEventListener("resize", handleResize);
+            return () => {
+                window.removeEventListener("resize", handleResize);
+            };
+        }
     }, []);
 
     // =========================================================================
     // RENDER FUNCTIONS
     // =========================================================================
-    const renderScreen = () => {
-        if (isLargeScreen) {
-            return (
-                <BuilderProvider>
-                    <DisplayProvider>
-                        <Wrapper>
-                            <Container type="grid" stretch>
-                                <ToastWrapper>
-                                    <Toasts />
-                                </ToastWrapper>
-                                <MainPanel />
-                                <SidePanel />
-                            </Container>
-                        </Wrapper>
-                    </DisplayProvider>
-                </BuilderProvider>
-            );
-        } else {
-            return (
-                <ErrorWrapper>
-                    <ErrorDisplay
-                        type="warning"
-                        title={"Screen size not supported"}
-                        description={
-                            "Expand to a bigger screen size or switch to desktop view to use the form builder."
-                        }
-                    />
-                </ErrorWrapper>
-            );
-        }
-    };
 
-    return renderScreen();
+    return isLargeScreen ? (
+        <BuilderProvider>
+            <DisplayProvider>
+                <Wrapper>
+                    <Container type="grid" stretch>
+                        <ToastWrapper>
+                            <Toasts />
+                        </ToastWrapper>
+                        <MainPanel />
+                        <SidePanel />
+                    </Container>
+                </Wrapper>
+            </DisplayProvider>
+        </BuilderProvider>
+    ) : (
+        <ScreenNotSupportedErrorDisplay />
+    );
```

Comment:
instead of conditionally hide the form builder, can we render the `ScreenNotSupportedErrorDisplay` above it?

because the current implementation is a destructive operation - all data regardless saved or unsaved are gone since everything is removed

Reply:
updated in the latest changes